js/classes/entities/midground/enemies/enemyBase.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * Class which provides basic functionality for enemy entities, including movement,
+ * taking damage, and block collisions.
+ * 
+ * @author Trae Claar
+ */
+class EnemyBase {
+    /**
+     * Constructor for an EnemyBase.
+     * 
+     * @param {Entity} enemy the enemy associated with this EnemyBase
+     * @param {Vector} pos the initial position of the enemy
+     * @param {number} size the size of the enemy
+     * @param {number} speed the speed of the enemy
+     * @param {number} health the initial health of the enemy
+     * @param {number} maxRoamDistance the maximum distance the enemy will wander 
+     *      from its initial position when roaming
+     * @param {function} onDeath callback function invoked when the enemy dies
+     * @param {string} stance the enemy's stance (use an EnemyBase member stance, 
+     *      e.g. EnemyBase.AggressiveStance)
+     * @param {number} reactionDistance the maximum distance that the enemy will 
+     *      react to Chad according to its stance and state
+     */
+    constructor(enemy, pos, size, speed, health, maxRoamDistance, onDeath, stance, reactionDistance) {
+        this.enemy = enemy;
+        enemy.pos = pos;
+        
+        enemy.state = "roam";
+        enemy.action = "idle";
+        enemy.size = size;
+        enemy.speed = speed; // *
+        enemy.health = health; // *
+        enemy.maxRoamDistance = maxRoamDistance; // *
+        enemy.yVelocity = 0;
+
+        enemy.boundingBox = new BoundingBox(pos, size);
+        enemy.lastBoundingBox = this.enemy.boundingBox;
+
+        this.maxHealth = health; 
+        this.onDeath = onDeath;
+        this.minRoamX = pos.x - maxRoamDistance;
+        this.targetX = pos.x;
+        this.stance = stance;
+        this.reactionDistance = reactionDistance ?? Camera.SIZE.x / 2 - size.x;
```

Comment:
That's called a nullish coalescing operator (fun name I know), and it returns the left operand if the left operand is not `null`/`undefined`, and returns the right operand otherwise.